src/content/datachannel/datatransfer/js/main.js

```diff
@@ -41,97 +43,81 @@ megsToSend.addEventListener('change', function(e) {
   }
 });
 
-function createConnection() {
+async function createConnection() {
   sendButton.disabled = true;
   megsToSend.disabled = true;
+  // Safari needs to trigger the media permission for RTCDataChannel to work
+  if (adapter.browserDetails.browser === 'safari') {
+    try {
+      console.log('Call getUserMedia() to trigger media permission request.');
+      const stream = await navigator.mediaDevices.getUserMedia({'audio': true, 'video': true});
```

Comment:
I don't see why this would be needed. The [associated issue has been marked as resolved](https://bugs.webkit.org/show_bug.cgi?id=174500) and this is not cross-browser. If it doesn't work, it's a bug.

Reply:
the samples don't use a stun server. Not sure about the state of MDNS in Safari...

Reply:
The linked issue has been resolved by the addition of mDNS. :man_shrugging:

Reply:
If there is a different workaround that fixes the problem in safari, please make a PR with that (or send me the details and I'll implement it). Thanks! :)

Reply:
What I'm saying is: There should be no need for a workaround if mDNS has been deployed in Safari. And if it hasn't yet, I would be strongly against adding that workaround because we shouldn't demonstrate such bad practice in official WebRTC examples.

Reply:
We got tons of workarounds in the code today. I looked for other ways to solve this but couldn't find any. The alternative is that this sample won't work in Safari.

Reply:
That is what I'm suggesting. Let it fail.

People will copy the sample code and believe it's fine to request media permissions for data only use cases because the official examples have done so. But it is definitely not fine. We should not encourage bad practices.

Reply:
Ok, let's do it like that. I'll update this PR tomorrow.